src/decode.rs

```diff
@@ -4542,40 +4553,28 @@ unsafe fn decode_b(
             case_set(bh4, &mut t.l, by4 as isize, set_ctx);
             case_set(bw4, &mut *t.a, bx4 as isize, set_ctx);
 
-            if (*f.frame_hdr).frame_type as libc::c_uint
-                & 1 as libc::c_int as libc::c_uint != 0
-            {
-                let r: *mut refmvs_block = &mut *(*(t.rt.r)
-                    .as_mut_ptr()
-                    .offset(
-                        ((t.by & 31 as libc::c_int) + 5 as libc::c_int + bh4
-                            - 1 as libc::c_int) as isize,
-                    ))
-                    .offset(t.bx as isize) as *mut refmvs_block;
-                let mut x: libc::c_int = 0 as libc::c_int;
-                while x < bw4 {
-                    (*r.offset(x as isize))
-                        .ref_0
-                        .ref_0[0 as libc::c_int as usize] = 0 as libc::c_int as int8_t;
-                    (*r.offset(x as isize)).bs = bs as uint8_t;
-                    x += 1;
+            if is_inter_or_switch(&*f.frame_hdr) {
+                let r: *mut refmvs_block = t
+                    .rt
+                    .r[((t.by & 31) + 5 + bh4 - 1) as usize]
+                    .offset(t.bx as isize);
```

Comment:
To be clear, were `.as_mut_ptr().offset(...)`s turned into `[...]` indexing (where there is a slice to start with), while bare `.offset(...)`s were kept the same (there's no slice, so indexing can't be done yet)?

Reply:
Got it, that make sense.  I started to come across all these instances, too, when I started cleaning up functions as well.